docs/testing-guide/concepts.md

```diff
@@ -0,0 +1,67 @@
+# Concepts
+
+The following sections provide an overview of the key concepts and features of the Algorand Python Testing framework.
+
+## Test Context
+
+The main abstraction to interact with the testing framework is the [`AlgopyTestContext`](algopy_testing.AlgopyTestContext). It creates an emulated Algorand environment that closely mimics AVM behavior relevant to unit testing the contracts and provides a Pythonic interface for interacting with the emulated environment.
+
+```python
+from algopy_testing import algopy_testing_context
+
+def test_my_contract():
+    # 1. Instantiating the test context
+    with algopy_testing_context() as ctx:
+        ... # your test code here
+
+    # 2. Alternatively, you can instantiate the test context manually
+    ctx = AlgopyTestContext()
+    ... # your test code here
+    ctx.reset()  # Reset the emulated environment
+```
+
+The `AlgopyTestContext` provides two methods for managing the state of the testing environment:
```

Comment:
The test context is quite lightweight to construct, perhaps if it would be better to just recommend creating a new instance rather than resetting/clearing the existing instance?

Reply:
Revamped the section - lmk what you think